Add new chapter - Browsers and the Web (#25)
pavpanchekha/emberfox: book/browsersandweb.md

competition for market share was channeled into very fast innovation and an
ever-expanding set of APIs and capabilities for the web, which we nowadays
refer to as _the Web Platform,_ not just the “World Wide Web”. This recognizes
the fact that the web is no longer a document viewing mechanism, but has
evolved into a fully realized computing platform and ecosystem[^25]. Given
these outcomes, it becomes clear that it’s not so relevant to know which
browser “won” or “lost” each of the browser “wars”. In both cases _the web won_
and was preserved and enhanced for the benefit of the world. In economics
terms, enforcing a standard set of APIs across browsers made the web platform a
_commodity_; instead of competing based on lock-in, browsers compete on
_performance_, and also _browser features_ that are not part of the web
platform - for example, tabbed UIs and search engine integration. Browser
development is also primarily funded by revenue from search engine
advertisements; a secondary typical funding motivation is to improve the
competitive position of an operating system or device owned or controlled by
the company building the browser[^26].

> An important important and interesting outcome of the second browser war was

Reviewer: ```suggestion
An important and interesting outcome of the second browser war was
```

that all mainstream browsers today (of which there are *many* more than three)
are based on _three open-source web rendering / JavaScript engines_: Chromium,

> Gecko and WebKit[^27]. Further, all three of these engines date back to the
> 1990s - i.e. almost to the beginning of the web, and two of them (Chromium and
> WebKit) have a common ancestral codebase. Gecko is an open-source descendant of
> code from Netscape. That this occurred is not an accident, and in fact tells us

Reviewer: ```suggestion
Gecko and WebKit[^27]. Since Chromium and
WebKit have a common ancestral codebase, while Gecko is an open-source descendant of
Netscape, all three date back to the 1990s---almost to the beginning of the web.
That this occurred is not an accident, and in fact tells us
```
